src/Haskell/Ide/Engine/Dispatcher.hs

```diff
@@ -23,7 +23,7 @@ data DispatcherEnv = DispatcherEnv
   , docVersionTVar     :: !(TVar (Map.Map Uri Int))
   }
 
-dispatcherP :: forall void. DispatcherEnv -> TChan PluginRequest -> IdeGhcM void
+dispatcherP :: DispatcherEnv -> TChan PluginRequest -> IdeGhcM ()
```

Comment:
Why is this needed? The old type is more descriptive, as it states the the function never "returns", i.e. loops forever.

The old type is also strictly more general than the new type(IdeGhcM () is a subtype of forall void. IdeGhcM void)